### PR TITLE
Improved/added kill and killall completions 

### PR DESCRIPTION
**Pull request by lordlycastle**
- Fixed kill signals completions
- Added OS X kill and killall completions
- Improved kill and killall completions for both systems

### CODE REVIEW

File: share/completions/kill.fish [hunk 1]
```diff
@@ -4,10 +4,21 @@ for i in $__kill_signals
 	set number (echo $i | cut -d " " -f 1)
 	set name (echo $i | cut -d " " -f 2)
 	complete -c kill -o $number -d $name
-	complete -c kill -o $name -d $name
-	complete -c kill -o s -x -a \"$number\tSend\ $name\ signal\"
-	complete -c kill -o s -x -a \"$name\tSend\ $name\ signal\"
+	complete -c kill -o $name
+	complete -c kill -s s -x -a "$number $name"
```
Comment (faho) [inline share/completions/kill.fish]: I still think having "number\t$name" and "$name" (without description) to be the way to go. The names are sort of descriptive (and we don't have a way to generate better descriptions) while the numbers aren't descriptive at all.

File: share/completions/killall.fish [hunk 2]
```diff
@@ -4,17 +4,48 @@ for i in $__kill_signals
...
 
 complete -c killall -xa '(__fish_complete_proc)'
-complete -c killall -s e -l exact -d 'Require an exact match for very long names'
-complete -c killall -s I -l ignore-case -d 'Do case insensitive process name match'
-complete -c killall -s g -l process-group -d 'Kill the process group to which the process belongs. The kill signal is only sent once per group, even if multiple processes belonging to the same process group were found'
-complete -c killall -s i -l interactive -d 'Interactively ask for confirmation before killing'
-complete -c killall -s u -l user -d 'Kill only processes the specified user owns. Command names are optional'
-complete -c killall -s w -l wait -d 'Wait  for  all  killed processes to die. killall checks once per second if any of the killed processes still exist and only returns if none are left.  Note that killall may wait forever if the signal was ignored, had no effect, or if the process stays in zombie state'
+
+if killall --verison > /dev/null ^ /dev/null
```
Comment (faho) [inline share/completions/killall.fish]: Typo - "--version", not "--verison".

Comment (faho) [inline share/completions/killall.fish]: Also, @zanchey mentioned in #2587 that calling `killall` on Solaris wasn't safe (there, `killall` just kills all processes on the system, so you don't want to execute it or it executes you).

So we should probably guard this with `uname`.

File: share/completions/kill.fish (hunk 1 above)
Comment (lordlycastle) [inline share/completions/kill.fish]: That's what it's generating right now.
screen shot 2015-12-19 at 14 41 53

File: share/completions/killall.fish (hunk 2 above)
Comment (lordlycastle) [inline share/completions/killall.fish]: > Also, @zanchey mentioned in #2587 that calling killall on Solaris wasn't safe (there, killall just kills all processes on the system, so you don't want to execute it or it executes you).

What's the output of `uname`for Solaris? And how would you stop user from executing that, what if they do want to intentionally do that? Perhaps if the system is solaris, then define a function with name `killall` that requires confirmation?

Comment (zanchey) [inline share/completions/killall.fish]: The problem is that if your completions run `killall --version` when they are loaded, then just typing `killall <TAB>` is enough to at best destroy your session and at worst hang the system.

`if test (uname) = SunOS` should do the trick.

**Comment by lordlycastle** [comment]
2e6aad2 should do it, simply returning when on Solaris.

**Comment by lordlycastle** [comment]
Right, I believe 5a298ea should do it. 

P.S. What is the usual protocol for fixing mistaken commits? Do you guys just usually just fix it and push a new one?

**Comment by faho** [comment]
>P.S. What is the usual protocol for fixing mistaken commits? Do you guys just usually just fix it and push a new one?

Usually we'll squash up PRs into one commit (because they are effectively one change). Though `git rebase --interactive` is amazingly powerful and can also only make the mistaken commits disappear, that's usually best done _after_ review.

**Comment by faho** [comment]
Merged as f23c2beb61075f9805347a84c8335ce1348e3d6d. Thanks!

**Pull request closed without merging**